### PR TITLE
Add the ability to search every drive and Steam Library on the computer

### PR DESCRIPTION
**Pull request by JPZV**
Two main features added:

1) It'll now search on every drive. E.g. Aside of looking on C:\\...\\Steam, it'll also search on D:\\...\\Steam, E:\\...\\Steam and go on (obviously only on available drives).

2) It'll also look at *:\Program Files (x86)\Steam\steamapps\libraryfolders.vdf file to get every Steam Library on the computer, not just *:\Program Files (x86)\Steam\steamapps\common\

For the second feature, I added the [Gameloop.Vdf](https://github.com/shravan2x/Gameloop.Vdf) library [(License)](https://github.com/shravan2x/Gameloop.Vdf/blob/master/LICENSE)

### CODE REVIEW

File: SunshineGameFinder/Program.cs [hunk 1]
```diff
@@ -50,16 +57,19 @@
         Logger.Log($"Could not find Sunshine Apps config at specified path: {sunshineAppsJson}", LogLevel.Error);
         return;
     }
-    var sunshineAppInstance = Newtonsoft.Json.JsonConvert.DeserializeObject<SunshineConfig>(File.ReadAllText(sunshineAppsJson));
+    var sunshineAppInstance = JsonConvert.DeserializeObject<SunshineConfig>(File.ReadAllText(sunshineAppsJson));
     var gamesAdded = 0;
-    foreach (var platformDir in gameDirs)
+    if (sunshineAppInstance == null)
```
Comment (JMTK) [inline SunshineGameFinder/Program.cs:62]: Add log here if it's null, although I would sooner expect an exception

File: SunshineGameFinder/Program.cs [hunk 2]
```diff
@@ -116,15 +126,43 @@
...
-        if (FileWriter.UpdateConfig(sunshineAppsJson, sunshineAppInstance))
+        if (platformDir.StartsWith(wildcatDrive))
+        {
+            foreach (var drive in logicalDrives)
+                ScanFolder(wildcatDriveLetter.Replace(platformDir, drive.Name, 1));
+        }
+        else
         {
-            Logger.Log($"Apps config is updated! {gamesAdded} games were added. Check Sunshine to ensure all games were added.", LogLevel.Success);
+            ScanFolder(platformDir);
```
Comment (JMTK) [inline SunshineGameFinder/Program.cs:161]: Can you merge in main so that this log isn't reverted?

File: SunshineGameFinder/Program.cs [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
 using System.CommandLine;
+using System.Text.RegularExpressions;
+
+// constants
+const string wildcatDrive = @"*:\";
+const string steamLibraryFolders = @"Program Files (x86)\Steam\steamapps\libraryfolders.vdf";
 
 // default values
-var gameDirs = new List<string>() { @"C:\Program Files (x86)\Steam\steamapps\common", @"C:\XboxGames", @"C:\Program Files\EA Games" };
+var gameDirs = new List<string>() { @"*:\Program Files (x86)\Steam\steamapps\common", @"*:\XboxGames", @"*:\Program Files\EA Games" };
```
Comment (JMTK) [inline SunshineGameFinder/Program.cs:14]: Not that it's a problem with your PR, but I just realized I didn't put Epic Games in if you wanted to go ahead and add that 
```*:\Program Files\Epic Games\```

File: SunshineGameFinder/Program.cs [hunk 4]
```diff
@@ -116,15 +126,43 @@
...
+        var file = new FileInfo(libraryFoldersPath);
+        if (!file.Exists)
+        {
+            Logger.Log($"{file.FullName} does not exist, skipping...", LogLevel.Warning);
+            continue;
+        }
+        var libraries = VdfConvert.Deserialize(File.ReadAllText(libraryFoldersPath));
+        foreach(var library in libraries.Value)
+        {
+            if (library is not VProperty libProp)
```
Comment (JMTK) [inline SunshineGameFinder/Program.cs:145]: Can you comment what exactly this would be skipping? I'm not super familiar with VDF

**Review by JMTK — CHANGES_REQUESTED** [review]
Thanks! I left some small feedback but looks good. I think it's super smart to do the VDF library thing, I didn't even know that existed

**Comment by JPZV** [comment]
Thanks! I made all the changes!

**Pull request merged**